packages/capabilities/src/provider.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Provider Capabilities
+ *
+ * These can be imported directly with:
+ * ```js
+ * import * as Provider from '@web3-storage/capabilities/provider'
+ * ```
+ *
+ * @module
+ */
+import { capability, DID, literal, struct } from '@ucanto/validator'
+import { equalWith, fail, equal } from './utils.js'
+import { top } from './top.js'
+
+export { top }
+
+/**
+ * Capability can only be delegated (but not invoked) allowing audience to
+ * derive any `provider/` prefixed capability for the agent identified
+ * by did:key in the `with` field.
+ */
+export const provider = top.derive({
+  to: capability({
+    can: 'provider/*',
+    with: DID,
+    derives: equalWith,
+  }),
+  derives: equalWith,
+})
```

Comment:
No longer need to do the `derive` stuff as ucanto now understands the `*` patterns natively. Also `equalWith` is obsolete because it's a default behavior.

```suggestion
export const provider = capability({
  can: 'provider/*',
  with: DID,
})
```

Reply:
iirc even `provider/*` automatically derives `provider/add`? Correct me if I'm wrong.
I removed this whole variable entirely in https://github.com/web3-storage/w3protocol/pull/462/commits/a199c11a4915b0858473f31e94072e881b8b737c